probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/mod.rs

```diff
@@ -236,18 +236,29 @@ fn main_try(args: &[OsString], offset: UtcOffset) -> Result<()> {
         };
         let format_options = FormatOptions::default();
         let loader = build_loader(&mut session, &path, format_options, image_instr_set)?;
-        run_flash_download(
+        let flash_info = run_flash_download(
             &mut session,
             &path,
             &download_options,
             &probe_options,
             loader,
             config.flashing.do_chip_erase,
         )?;
-    }
+
+        if flash_info.entry_point_in_ram {
+            session.ram_flash_start(flash_info.entry_point.unwrap())?;
+        }
+
+        flash_info.entry_point_in_ram
+    } else {
+        false
+    };
 
     let core_id = rtt::get_target_core_id(&mut session, &path);
-    if config.reset.enabled {
+
+    if ram_flash {
+        session.core(core_id)?.run()?;
```

Comment:
Please move "ram_flash_start" (after renaming to boot_from_ram) to here.

Reply:
Well, I see why it's not there... But separating "start the process" and "start the core" looks weird